pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -86,6 +97,12 @@ func (c NodeController) sync() error {
 		if found {
 			newTargetNodeStates = append(newTargetNodeStates, originalOperatorStatus.NodeStatuses[i])
 		} else {
+			// Prune installer pods we created for this node. In case these are "pending" they will never go to running
+			// as their nodeName won't exists.
+			if err := c.pruneInstallerPodsForRemovedNode(nodeState.NodeName); err != nil {
+				c.eventRecorder.Warningf("MasterNodePruningFailed", "Failed to prune pods for removed master node %s: %v", nodeState.NodeName, err)
+				return err
```

Comment:
@deads2k not sure if this is worth erroring out or should be just best-effort.

Reply:
I think this should be best effort.